File: githubAppUpdate/src/main/kotlin/info/hannes/github/IGithub.java
package info.hannes.github;

import java.util.List;

import info.hannes.github.model.GithubVersion;
import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Headers;
import retrofit2.http.Path;

public interface IGithub {

    @GET("/repos/{githubUser}/{githubRepo}/releases")
    @Headers("Accept: application/json")
    Call<List<GithubVersion>> getGithubVersions(
            @Path("githubUser") String githubUser,
            @Path("githubRepo") String githubRepo);

}

File: githubAppUpdate/src/main/java/info/hannes/github/IGithub.kt
package info.hannes.github

import info.hannes.github.model.GithubVersion
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Headers
import retrofit2.http.Path

interface IGithub {
    @GET("/repos/{githubUser}/{githubRepo}/releases")
    @Headers("Accept: application/json")
    fun getGithubVersions(
        @Path("githubUser") githubUser: String,
        @Path("githubRepo") githubRepo: String
    ): Call<List<GithubVersion>>
}
